cmd/skopeo/inspect.go

```diff
@@ -106,6 +108,24 @@ func (opts *inspectOptions) run(args []string, stdout io.Writer) (retErr error)
 		}
 	}()
 
+	unparsedImage := image.UnparsedInstance(src, nil)
+	if opts.checkPolicy {
+		policyContext, err := opts.global.getPolicyContext()
+		if err != nil {
+			return fmt.Errorf("Error loading trust policy: %v", err)
+		}
+		defer func() {
+			if err := policyContext.Destroy(); err != nil {
+				retErr = noteCloseFailure(retErr, "tearing down policy context", err)
+			}
+		}()
+
+		// Be paranoid and fail if either return value indicates so.
+		if allowed, err := policyContext.IsRunningImageAllowed(ctx, unparsedImage); !allowed || err != nil {
```

Comment:
This should also be inside `retry.IfNecessary` because it contacts remote servers to fetch the signatures.